ui/components/app/flask/snaps-authorship-pill/snaps-authorship-pill.js

```diff
@@ -15,8 +15,10 @@ const snapIdPrefixes = ['npm:', 'local:'];
 
 const SnapsAuthorshipPill = ({ snapId, version, className }) => {
   // @todo Use getSnapPrefix from snaps-skunkworks when possible
-  const snapPrefix = snapIdPrefixes.find((prefix) => snapId.startsWith(prefix));
-  const packageName = snapId.replace(snapPrefix, '');
+  const snapPrefix = snapIdPrefixes.find((prefix) =>
+    snapId?.startsWith(prefix),
```

Comment:
Ok, I think I know what the issue is. There's a race condition if I'm not mistaken, when you hit cancel the request gets deleted and hence there is no reference to request.snapId. Things work with snap-install because when installing a snap for the first time, the metadata is added before the permissions are requested so we have access to snapId directly from persisted state, whereas when you hit cancel the request that supplies the snapId is being deleted right away. I mentioned this issue in my slack post, we can leave this like it is and I can take care of it in a subsequent PR.